tests/test_unit/test_sample_data.py

```diff
@@ -127,28 +134,79 @@ def test_list_datasets(valid_sample_datasets):
-
-        assert getattr(ds, "fps", None) == sample["fps"]
-
-        frame_path = getattr(ds, "frame_path", None)
-        video_path = getattr(ds, "video_path", None)
-
-        if sample["frame_file"]:
-            assert frame_path.split("/")[-1] == sample["frame_file"]
-        else:
-            assert frame_path is None
-
-        if sample["video_file"] and with_video:
-            assert video_path.split("/")[-1] == sample["video_file"]
-        else:
-            assert video_path is None
-
-    # Test with an invalid file
-    with pytest.raises(ValueError):
-        fetch_dataset("nonexistent_file")
+def test_fetch_dataset(sample_name, sample_metadata, with_video):
+    """Test fetch_dataset for each sample dataset with/without video."""
+    ds = fetch_dataset(sample_name, with_video=with_video)
+    assert isinstance(ds, Dataset)
+
+    # Check fps attribute
+    assert getattr(ds, "fps", None) == sample_metadata["fps"]
+
+    # Check frame_path
+    frame_path = getattr(ds, "frame_path", None)
+    if sample_metadata["frame_file"]:
+        assert frame_path is not None
+        assert frame_path.split("/")[-1] == sample_metadata["frame_file"]
+    else:
+        assert frame_path is None
+
+    # Check video_path
+    video_path = getattr(ds, "video_path", None)
+    if sample_metadata["video_file"] and with_video:
+        assert video_path is not None
+        assert video_path.split("/")[-1] == sample_metadata["video_file"]
+    else:
+        assert video_path is None
+
+
+@pytest.mark.parametrize(
+    "sample_name, expected_exception",
+    [
+        ("nonexistent_file", ValueError),
+        ("TRex_five-locusts.zip", NotImplementedError),
+    ],
+    ids=["invalid_file", "TRex_folder_zip"],
+)
+def test_fetch_dataset_invalid(sample_name, expected_exception):
+    with pytest.raises(expected_exception):
+        fetch_dataset(sample_name)
+
+
+@pytest.mark.parametrize(
+    "sample_name",
+    [
+        list(SAMPLE_DATASETS.keys())[0],
+        "TRex_five-locusts.zip",
+    ],
+    ids=[
+        "poses_in_single_file",
+        "poses_in_zipped_folder",
+    ],
+)
+def test_fetch_dataset_paths(sample_name):
+    """Test that the returned pose paths points to correct location.
+
+    If the pose files are in a zipped folder, the path should point to the
+    unzipped folder, otherwise it should point to the file itself.
+    """
+    paths = fetch_dataset_paths(sample_name)
+    poses_path = Path(paths["poses"])
+
+    if sample_name.endswith(".zip"):
+        # If the sample is a zip file,
+        # the path should point to the unzipped folder
+        assert poses_path.is_dir()
+        assert poses_path.name == sample_name.replace(".zip", "")
+        assert len(list(poses_path.iterdir())) > 1
+    else:
+        # If the sample is a single file,
+        # the path should point to the file itself
+        assert poses_path.is_file()
+        assert poses_path.name == sample_name
```

Comment:
This will cover all 5 test samples we have, which I suppose isn't a bad idea
```suggestion
def test_fetch_dataset_paths(sample_dataset):
    """Test that the returned pose paths points to correct location.

    If the pose files are in a zipped folder, the path should point to the
    unzipped folder, otherwise it should point to the file itself.
    """
    sample_name, _ = sample_dataset
    paths = fetch_dataset_paths(sample_name)
    data_path = Path(paths.get("poses", paths.get("bboxes")))

    if sample_name.endswith(".zip"):
        # If the sample is a zip file,
        # the path should point to the unzipped folder
        assert data_path.is_dir()
        assert data_path.name == sample_name.replace(".zip", "")
        assert len(list(data_path.iterdir())) > 1
    else:
        # If the sample is a single file,
        # the path should point to the file itself
        assert data_path.is_file()
        assert data_path.name == sample_name
```